src/Sparrow.Server/LowMemory/MemoryInformation.cs

```diff
@@ -335,6 +343,18 @@ internal static MemoryInfoResult GetMemoryInfo(SmapsReader smapsReader = null, b
             }
         }
 
+        public static long GetTotalScratchAllocatedMemory()
+        {
+            long totalScratchAllocated = 0;
+            foreach (var scratchGetAllocated in DirtyMemoryObjects)
+            {
+                if (scratchGetAllocated.Value != null)
```

Comment:
Use: `totalScratchAllocated += scratchGetAllocated.Value?.Invoke()`

Right now, you have a race condition here that will kill us with NRE in dedicated thread